python/lsst/ts/mtmount/tma_telemetry_config_parser.py

```diff
@@ -0,0 +1,599 @@
+        config.optionxform = str  # preserve key case
+        config.read(tma_config_path)
+        return config
+
+    def make_basic_fields(
+        self, section: configparser.SectionProxy
+    ) -> tuple[int, FieldsType]:
+        """Process one section (topic).
+
+        Do not consolidate array-like values.
+
+        Parameters
+        ----------
+        section : `configparser.SectionProxy`
+            One section (telemetry topic) of the TMA config file.
+
+        Returns
+        -------
+        topic_id_field_info
+            A tuple of:
+
+            * topic_id: integer topic ID
+            * fields: a dict of field name: FieldInfo
+
+        Notes
+        -----
+        Add the ``timestamp`` field, which is not in the TMA telemetry config
+        (except for internal use in a few cases; ignore that one).
+        """
+        topic_sal_name = section.name[0].lower() + section.name[1:].replace(" ", "")
+        topic_attr_name = f"tel_{topic_sal_name}"
+        topic_id: int | None = None
+
+        # Dict of field_name: FieldInfo
+        fields = dict()
+        name = None
+        units = None
+        description = None
+        for i, (key, value) in enumerate(section.items()):
+            if not value.startswith('"'):
+                raise RuntimeError(
+                    f'All values must start with ": key={key}; value={value!r}; '
+                    f"section={section.name}"
+                )
+            value = value[1:-1]
+            key_kind = key.rsplit(".", 1)[-1]
+            if i == 0:
+                if key != "TopicID":
+                    raise RuntimeError(
+                        "The first line of each section must be TopicID: "
+                        f"key={key}; section={section.name}"
+                    )
+                topic_id = int(value)
+            elif key in {
+                "TopicFrequencyMultiple50ms",
+                "Int64 Array Telemetry Data.<size(s)>",
+                "INT32 Telemetry Data.<size(s)>",
+                "String Array Telemetry Data.<size(s)>",
+            }:
+                pass
+            elif key_kind == "<size(s)>":
+                pass
+            elif key_kind == "url":
+                pass
+            elif key_kind == "Unit":
+                units = value
+            elif key_kind == "TCP_PublishName":
+                name = value
+            elif key_kind == "Comments":
+                description = value
+            elif key_kind == "TCP_Publish":
+                if None in (name, units, description):
+                    raise RuntimeError(
+                        f"{name=!r}, {units=!r}, and {description=!r} cannot be empty; "
+                        f"key={key}; section {section.name}"
+                    )
+                try:
+                    # Ignore psp://192.168.209.10/PXIComm_NSV/CCW timestamp
+                    # which Tekniker insists have TCP_Publish = "TRUE"
```

Comment:
I am not sure it is allowed to expose the vendor's name (Tekniker) or not, which is supposed to be confidential.

Reply:
This should be a matter of public record. I will risk it.